qemu/tests/disk_extension.py

```diff
@@ -0,0 +1,136 @@
+    5) Boot vm with loop device as data disk.
+    6) Access  guest vm and execute dd operation on the data disk.
+     the IO size is same as the loop device virtual disk size.
+    7) Verify vm status is io-error status in hmp.
+    8) Continue to increase disk size of the raw image file and update the loop device.
+    9) Verify vm status whether in expected status:
+      if the raw image file size is smaller than loop device virtual disk size,
+      it is in io-error status,Otherwise it is in running status.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def prepare_tmpfs_folder(dirname):
+        logging.debug("Ready to create tmpfs folder on %s" % dirname)
+        cmd = "umount -l {0};rm -rf {0};mkdir -p {0};".format(dirname)
+        process.system_output(cmd, ignore_status=True, shell=True)
+        cmd = "mount -t tmpfs -o rw,nosuid,nodev,seclabel tmpfs {}".format(dirname)
+        process.system_output(cmd, shell=True)
+
+    def setup_loop_device_on_image(backend_img, device_img):
+        if backend_img.image_format != "raw":
+            test.error("Only support attaching loop device on raw format image.")
+            return
+        backend_img.create(backend_img.params)
+        backend_filename = backend_img.image_filename
+        loop_device_name = device_img.image_filename
+        logging.debug("Ready to create loop device image {} on {}".format(loop_device_name, backend_filename))
+        cmd = "losetup -d {}".format(loop_device_name)
+        process.system_output(cmd, ignore_status=True, shell=True)
+        cmd = "losetup {0} {1} && chmod 666 {0}".format(loop_device_name, backend_filename)
+        process.system_output(cmd, shell=True)
+        device_img.create(device_img.params)
+        logging.debug("Finish creating loop device image {} on {}".format(loop_device_name, backend_filename))
+
+    def update_loop_device_backend_size(backend_img, device_img, size, unit="M"):
+        cmd = "qemu-img resize -f raw {} {}".format(backend_img.image_filename, str(size) + unit)
+        process.system_output(cmd, shell=True)
+        # update loop device
+        cmd = "losetup -c {0}".format(device_img.image_filename)
+        process.system_output(cmd, shell=True)
+
+    error_context.context("Get required parameter from the config file",
+                          logging.info)
+    current_size = int(params["begin_size"])
+    max_size = int(params["max_size"])
+    increment_size = int(params["increment_size"])
+    guest_cmd = params["guest_cmd"]
+    loop_device_backend_img_param = params.object_params('stg0')
+    loop_device_img_param = params.object_params('stg1')
+    drive_format = loop_device_img_param["drive_format"]
+    image_aio = loop_device_img_param["image_aio"]
+    if loop_device_backend_img_param["image_format"] != "raw":
```

Comment:
is it better to cancel the test if the backend image format in config isn't raw?